classes/Product.php

```diff
@@ -8262,4 +8262,20 @@ protected function updateDefaultShop(): void
             $this->id_shop_default = $firstAssociatedShop;
         }
     }
+
+    /**
+     * Check if a product is active.
+     *
+     * @param int $id_product The ID of the product to check.
+     * @return bool Returns true if the product is active, false otherwise.
+     */
+    public static function isActive(int $id_product): bool
+    {
+        return (bool) Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue(
+            'SELECT product_shop.`active`
+            FROM `' . _DB_PREFIX_ . 'product` p
+            ' . Shop::addSqlAssociation('product', 'p') . '
+            WHERE p.`id_product` = ' . $id_product
+        );
+    }
```

Comment:
I wonder if we should do something more generic in object model?

Reply:
this can be done in another PR of course

Reply:
Indeed, that would be interesting

Reply:
What about this in object model? :-)

```
    /**
     * Check if object is active.
     *
     * @param int $id_entity
     *
     * @return bool
     */
    public static function isActive($id_entity)
    {
        $className = static::class;

        // Object must have a variable called 'active'
        if (!isset(static::$definition['fields']['active'])) {
            throw new PrestaShopException('property "active" is missing in object ' . $className);
        }

        // Construct the object using definition classname
        $object = new $className((int) $id_entity);

        // Check if object exists
        if (!Validate::isLoadedObject($object)) {
            return false;
        }

        return (bool) $object->active;
    }
```

Reply:
I also a draft ready : 
`/**
     * Checks if the given object is active.
     *
     * @param int $objectId The ID of the object to check.
     * @return bool Returns true if the object is active, false otherwise.
     */
    public static function isActive(int $objectId): bool
    {
        if (!property_exists(static::class, 'active')) {
            return false;
        }
        $def = (static::class)::$definition;
        $activeDef = $def['fields']['active'];
        $tableName = $def['table'];
        if (!empty($activeDef['shop'])) {
            $assoName = $def['table'] . '_shop';
            return (bool) Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue(
                'SELECT ' . $assoName . '.`active`
            FROM `' . _DB_PREFIX_ . $tableName . '` t
            ' . Shop::addSqlAssociation($tableName, 't') . '
            WHERE t.`' . $def['primary'] . '` = ' . $objectId
            );
        }
        return (bool) Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue(
            'SELECT t.`active`
            FROM `' . _DB_PREFIX_ . $tableName . '` t
            WHERE t.`' . $def['primary'] . '` = ' . $objectId
        );
    }`
    Yours is using the already loaded object and it's a nice thing to do instead of executing an useless query. However will return false if object is not loaded. IMO that is wrong, because developer doesn't know if object is false, or if object is not loaded. I'd rather mix our two versions, and execute the query in case the object is not loaded.